group/views.py

```diff
@@ -205,6 +214,73 @@ def edit(request, group_id):
                 if form.is_valid():
                     modified_group = form.save()
                     logger.info('Group: Modified group %s!' % modified_group.id)
-                    return HttpResponseRedirect('/group/detail/%s' % modified_group.id)
+                    return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': modified_group_id}))
         else:
             raise PermissionDenied
+
+##Announce
+@login_required
+def add_announce(request, group_id):
+    group = get_group(group_id)
+
+    if has_group_ownership(request.user, group) or has_group_coownership(request.user, group):
+        if request.method == 'POST':
+            form = AnnounceForm(request.POST)
+            if form.is_valid():
+                new_announce = form.save()
+                group.announce.add(new_announce)
+                logger.info('Announce: User %s add Announce %s!' % (request.user.username, new_announce.id))
+                return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+            else:
+                return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+    else:
+        raise PermissionDenied
+
+@login_required
+def delete_announce(request, announce_id, group_id):
+    group = get_group(group_id)
+
+    if has_group_ownership(request.user, group) or has_group_coownership(request.user, group):  
+        get_announce(announce_id).delete()
+        return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+    else:
+        raise PermissionDenied
+
+@login_required
+def edit_announce(request, announce_id, group_id, redirect_id):
+    announce = get_announce(announce_id)
+    group = get_group(group_id)
+    user_is_owner = has_group_ownership(request.user, group)
+    user_is_coowner = has_group_coownership(request.user, group)
+    if user_is_owner or user_is_coowner or request.user.has_admin_auth():
+        if request.method == 'GET':
+            announce_dic = model_to_dict(announce)
+            form = AnnounceForm(initial=announce_dic)
+            return render_index(
+                request,'group/editAnnounce.html', {
+                    'form':form,
+                    'group_id': group.id,
+                    'announce_id': announce_id,
+                    'user_is_owner': user_is_owner,
+                    'user_is_coowner': user_is_coowner,
+                    'user_has_admin_auth': request.user.has_admin_auth(),
+                })
+        if request.method == 'POST':
+            form = AnnounceForm(request.POST, instance=announce)
+            if form.is_valid():
+                modified_announce = form.save()
+                logger.info('redirect_id: %s!' % redirect_id)
+                if redirect_id == '1':
+                    return HttpResponseRedirect(reverse('group:detail', kwargs={'group_id': group_id}))
+                elif redirect_id == '0' :
```

Comment:
Ah, That's because redirect_id is used to distinct whether it should be redirect to "detail" page or "view_all" page. Since "view_all" is not  included in this PR, you will not find any place assign this variable to 0.